runtime/lua/vim/fs.lua

```diff
@@ -73,10 +73,14 @@ function M.basename(file)
 end
 
 ---@private
-local function join_paths(...)
+---@param ... string
+---@return string
+function M._join_paths(...)
```

Comment:
Any reason we haven't exposed this as `fs.join()` ?

Reply:
(nvim-treesitter also implements path handling; would be great to replace that with `vim.fs` as much as possible)